SimPEG/Problem.py

```diff
@@ -6,45 +6,96 @@
 from . import Maps
 from .Fields import Fields, TimeFields
 from . import Mesh
+from . import Props
+import properties
 
 
 Solver = Utils.SolverUtils.Solver
 
 
-class BaseProblem(object):
-    """
-        Problem is the base class for all geophysical forward problems in SimPEG.
+class BaseProblem(Props.HasModel):
+    """Problem is the base class for all geophysical forward problems
+    in SimPEG.
     """
 
-    counter = None   #: A SimPEG.Utils.Counter object
+    _depreciate_maps = False
+    _depreciate_main_map = None
+
+    #: A SimPEG.Utils.Counter object
+    counter = None
+
+    #: A SimPEG.Survey Class
+    surveyPair = Survey.BaseSurvey
+
+    #: A SimPEG.Map Class
+    mapPair = Maps.IdentityMap
+
+    #: A SimPEG Solver class.
+    Solver = Solver
+
+    #: Solver options as a kwarg dict
+    solverOpts = {}
+
+    #: A SimPEG.Mesh instance.
+    mesh = None
+
+    def __init__(self, mesh, **kwargs):
+        if 'mapping' in kwargs:
+            import warnings
```

Comment:
if we are going to hard-depreciate, would it be best it this were an error rather than a warning - the code will fail later on regardless, so it might be best to fail here. Thoughts @rowanc1?